### PR TITLE
feat(eslint): Add NewRelic Eslint plugin and format/fixes

### PR DESCRIPTION
**Pull request by devfreddy**

### CODE REVIEW

File: nerdlets/customer-journey/StatCell.js [hunk 1]
```diff
@@ -18,14 +18,12 @@ export default class StatCell extends React.Component {
     step: PropTypes.object.isRequired,
     column: PropTypes.object.isRequired,
     config: PropTypes.object.isRequired,
-    timeRange: PropTypes.object.isRequired,
```
Comment (danielgolden) [inline nerdlets/customer-journey/StatCell.js]: I think this conflicts with my VS code config (however that's done). By that I mean, when I have "format on save" enabled, if I don't have a comma at the end of the last item in an object, it adds it for me.

Comment (danielgolden) [reply nerdlets/customer-journey/StatCell.js]: Is it just me or is it in yours too?

Comment (devfreddy) [reply nerdlets/customer-journey/StatCell.js]: I've experienced this before, I forget exactly what caused it... I think eslint was doing one thing, and prettier was doing the other, and both were firing.

With the direction we're headed, you should disable having Prettier run on save, and just have eslint run on save. We're going to configure the nerdpacks to utilize some ESlint rules that will _call_ prettier. Then its only called once.

So, another way of putting it is:
- Turn off the VSCode wide "Format On Save"
- Turn on the Plugin specific "ESLint Format On Save" option

**Comment by devfreddy** [comment]
@danielgolden Some of this is autofixes from the eslint plugin, most of the rest are "easy" fixes to satisfy some lint rules that error out when you run `eslint-check` - mostly fall into 3 categories:
- equality operators, changing `==` to `===`
- removing unused imports and variables etc.
- adding propTypes to functions/classes etc.

**Pull request merged**